Woof/Woof/Model/LocalStorage/KeyValueStorage.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+/// A persistent storage designed for managing a collection of raw data (a byte buffer in memory).
+/// This data is stored and retrieved using a key that uniquely identifies itself.
+/// The key is used to find the data within the storage.
+class KeyValueStorage {
+    /// Instance of persistent storage with the specified database name.
+    private let storage = UserDefaults(suiteName: "com.foundation-iron.iOS.Woof")
+    /**
+      Saves the data associated with the specified key.
+
+      - Parameters:
+         - data: The data to save as a byte buffer in memory.
+         - key: The key associated with the value to store.
+      - Returns: `true` if data was successfully saved, otherwise returns `false`.
+     */
+    @discardableResult func save(data: Data, for key: String) -> Bool {
+        storage?.set(data, forKey: key)
+        let result = storage?.object(forKey: key) as? Data
+        return result != nil ? true : false
```

Comment:
Not sure I understood the logic here.
Can we just do smth like that:
```suggestion
        guard let storage else { return false }
        storage.set(data, forKey: key)
        return true
```
?